test/integration/builder/transaction.js

```diff
@@ -415,6 +409,31 @@ module.exports = function(knex) {
       });
     });
 
+    it('#2213 - should not evaluate a Transaction container until all previous siblings have completed', async function() {
+      const TABLE_NAME = 'test_sibling_transaction_order';
+      await knex.schema.dropTableIfExists(TABLE_NAME);
+      await knex.schema.createTable(TABLE_NAME, function(t) {
+        t.string('username');
+      });
+
+      await knex.transaction(async function(trx) {
+        await Promise.all([
+          trx.transaction(async function(trx1) {
+            // This delay provides `trx2` with an opportunity to run first.
+            await delay(200);
+            await trx1(TABLE_NAME).insert({ username: 'bob' });
+          }),
+          trx.transaction(async function(trx2) {
+            const rows = await trx2(TABLE_NAME);
+
+            // Even though `trx1` was delayed, `trx2` waited patiently for `trx1`
+            // to finish.  Therefore, `trx2` discovers that there is already 1 row.
+            expect(rows.length).to.equal(1);
+          }),
+        ]);
+      });
+    });
```

Comment:
The other test case for #2213 would still succeed for `oracledb` even though the transaction ordering had not been properly enforced.

So, this raises an important question: should this transaction ordering behavior apply to `oracledb` as well?  If so, then I think the current PR is good as-is.  Otherwise, we might need to do some additional refactoring to extract this behavior even further.